### ISSUE TITLE
[css-sizing-4] The priority of different minimum and maximum sizes

### ISSUE THREAD

**Issue by cathiechen** [w3c/csswg-drafts#7461]
Now, we have different kinds of minimum/maximum sizes:
* The Definite minimum/maximum sizes
* [Transferred Min/Max Sizes](https://drafts.csswg.org/css-sizing-4/#aspect-ratio-size-transfers)
* [Automatic Content-based Minimum Sizes](https://drafts.csswg.org/css-sizing-4/#aspect-ratio-minimum)

1. Per [1], the "Definite minimum/maximum sizes" has higher priority than the "Automatic Content-based Minimum Sizes".
```
<!DOCTYPE html>
<style>
    #target {
        background: green;
        height: 100px;
        aspect-ratio: 1 / 1;
        max-width: 30px;
    }
</style>
<div id="target">
    <div style="width:200px;"></div>
</div>
```
The definite maximum width is 30px, the Automatic Content-based Minimum width is 200px, so the calculated width of target should be 30px, IIUC.
Tested in Chromium, WebKit and Gecko, the calculated width of target is 30px. I think we need a test for it if we have an agreement here. 

2. Per [2], the "Definite minimum/maximum sizes" has higher priority than the "Transferred Min/Max Sizes".

```
<!DOCTYPE html>
<style>
    #target {
        background: green;
        height: 100px;
        aspect-ratio: 1 / 1;
        max-height: 50px;
        max-width: 30px;
    }
</style>

<div id="target"></div>
```
The "Definite maximum width" is 30px, the "Transferred Max width" is 50px, so the width of target should be 30px.
WebKit is 50px, Chromium and Gecko is 30px. Need a WPT test for it here.

3. It seems we have not defined the priority of "Transferred Min/Max Sizes" and "Automatic Content-based Minimum Sizes" explicitly.
```
<!DOCTYPE html>
<style>
    #target {
        background: green;
        height: 100px;
        aspect-ratio: 1 / 1;
        max-height: 30px;
    }
</style>

<div id="target">
    <div style="width:200px;"></div>
</div>
```
The "Transferred Max width" is 30px, and the "Automatic Content-based Minimum width" is 200px.
The behavior of the engines are consistent, all are 200px.
I do not have a strong opinion here, if the agreement is 200px, need a WPT test for it.

[1] https://drafts.csswg.org/css-sizing-4/#aspect-ratio-minimum
[2] https://drafts.csswg.org/css-sizing-4/#aspect-ratio-size-transfers

**Comment by cathiechen** [comment]
Hi Dave,
Thanks for the swift reply!

> 1. Explicit min-size beats automatic min-size, but auto min-size beats explicit max-size, which is why all engines say 200px. I think min beating max comes all the way from CSS2, due to the ordering of steps 2 and 3 in https://drafts.csswg.org/css2/#min-max-widths. Maybe something in css-sizing-3 reorganizes or restates that, but I didn't see it at a quick glance.

Thanks! Now I understand it:)

In https://drafts.csswg.org/css-sizing-4/#aspect-ratio-minimum, it says:
> In order to avoid unintentional overflow, the [automatic minimum size](https://drafts.csswg.org/css-sizing-3/#automatic-minimum-size) in the [ratio-dependent axis](https://drafts.csswg.org/css-sizing-4/#ratio-dependent-axis) of a box with a [preferred aspect ratio](https://drafts.csswg.org/css-sizing-4/#preferred-aspect-ratio) that is neither a [replaced element](https://drafts.csswg.org/css-display-3/#replaced-element) nor a [scroll container](https://drafts.csswg.org/css-overflow-3/#scroll-container) is **its [min-content size](https://drafts.csswg.org/css-sizing-3/#min-content) capped by its [maximum size](https://drafts.csswg.org/css-sizing-3/#max-width)**.

For the case 1,  if the min-content(200px) is capped by the maximum size(30px), so the auto min-size is 30px. Maybe we need to reword a little bit here?

**Comment by davidsgrogan** [comment]
No, I think my first comment was wrong. You're right, the spec calls for explicit max sizes to beat auto min sizes. It appears that Blink just doesn't do that. [It sets the min-width to min-intrinsic](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/layout/ng/ng_length_utils.cc;drc=917850e016efe08e27c61eaa03e6fc27200e5be1;l=531) and never again checks for an explicit max-width.

**Comment by Loirooriol** [comment]
The automatic minimum size is clamped by maximums in grid and flex layout, so I think it should be the same in the `aspect-ratio` case (as the spec already says).
